netbsd patch: core/sys/posix

sibnick opened the pull request:
Sync patch with current ldc source code

Commit:
netbsd patch: core/sys/posix

Commit:
netbsd patch

UplinkCoder committed:
ctfe-code is often annotated with pure @safe we want to print anyway

TurkeyMan committed:
Added some missing UUID's.

MartinNowak committed:
Merge pull request #1666 from TurkeyMan/update_uuid

Added some missing UUID's

jpf91 committed:
ARM: Explicitly specify required FPU mode in Fiber ASM

The threadASM code requires a vfp or newer unit for the vpush
and vpop instructions. There's exaclty one older floating point ARM
unit: The fpa unit. This unit is not supported in GCC but it is
supported in binutils.

This causes the following problem: When GCC is not explicitly configured
with --with-fpu it will not pass fpu flags to binutils as. So binutils
will use the binutils standard mode which is fpa! This then causes
compilation errors.

So in the end even though the gcc preprocessor claims vfp support via
__ARM_PCS_VFP we still have to request vfp explicitly in the ASM.

Note that this is not a restriction: As GCC doesn't support FPA
we can't compile phobos in FPA mode anyway. Newer floating point
units are fully compatible with vfp, so we can specify .fpu vfp
and still compile phobos with support for newer FPUs.

WalterBright committed:
add core.simd.prefetch declaration

MartinNowak committed:
fix documentation

- remove extra column of 0s (guess that was writeFetch before)
- fix THDR->THEAD

MartinNowak committed:
Merge pull request #1669 from WalterBright/prefetch

add core.simd.prefetch declaration

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

MartinNowak committed:
Merge pull request #1670 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable

dnadlinger committed:
Merge pull request #1667 from jpf91/fiberARM

ARM: Explicitly specify required FPU mode in Fiber ASM

Burgos committed:
Fix issue 16470: Segfault with negative array length

If the array length is too large, and the element
size small enough, the overflow might go undetected
in the check while multiplying array size and element
size, but it can later manifest when adding padding, etc.

https://issues.dlang.org/show_bug.cgi?id=16470

Burgos committed:
Check if the array allocation succeded

Burgos committed:
Add tests for "Out of memory exception"

Burgos committed:
Add linux-specific `core.sys.linux.sys.file` module

This adds core.sys.linux.file module which contains
flock function definition and values of the parameters
to pass.

jmdavis committed:
Make FD_* functions for select pure.

This fixes it so that the FD_* functions (which are macros in C/C++) are
pure on POSIX systems. The Windows versions (in
core.sys.windows.winsock2) are already pure.

dnadlinger committed:
Merge pull request #1680 from jmdavis/pure_select_helpers

Make FD_* functions for select pure.

CyberShadow committed:
core.time: Make private helper private

schveiguy committed:
Merge pull request #1682 from CyberShadow/pull-20161023-150220

[trivial] core.time: Make private helper private

nordlow committed:
Qualify C-style memory allocations functions as pure

MartinNowak committed:
remove __ctfeWriteln

- just append a newline before the call, or format the string

MartinNowak committed:
Merge pull request #1654 from UplinkCoder/__ctfeWriteln_fix

ctfe-code is often annotated with pure @safe we want to print anyway

pineapplemachine committed:
Added private qualifier to one-off unittest helper isnan in object.d

andralex committed:
Merge pull request #1683 from nordlow/pure-stdlib-alloc

Qualify C-style memory allocation functions as pure

yebblies committed:
Merge pull request #1686 from pineapplemachine/private-object-isnan

Added private qualifier to one-off unittest helper isnan in object.d

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

rainers committed:
do not enforce specific MS C runtime

rainers committed:
move new switch into separate line to not cause merge conflict with the auto tester

rainers committed:
Merge pull request #1689 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable

somzzz committed:
fix issue 1761

somzzz committed:
Merge branch 'master' into issue_1761

somzzz committed:
removed whitespaces

somzzz committed:
Test cov

somzzz committed:
fixed @ comments

andralex committed:
Merge pull request #1692 from somzzz/issue_1761

fix issue 1761 - TypeInfo.toString for function types always indicates no-parameter function

WalterBright committed:
fix Issue 16558 - [Mir] Generic unaligned load/store like (like LDC loadUnaligned and storeUnaligned)

andralex committed:
Merge pull request #1693 from WalterBright/loadUnaligned

fix Issue 16558 - [Mir] Generic unaligned load/store like (like LDC l…

WalterBright committed:
fix Issue 16743 - Intrinsic recognition sometimes fails if a software implementation is available

somzzz committed:
fix issue 5995

nemanja-boric-sociomantic committed:
Add bindings for POSIX iconv

somzzz committed:
unittest

somzzz committed:
unittest

leandro-lucarella-sociomantic committed:
Add missing POSIX <libgen.h> header

andralex committed:
Merge pull request #1695 from WalterBright/fix16743

fix Issue 16743 - Intrinsic recognition sometimes fails if a software…

nemanja-boric-sociomantic committed:
Add bindings for getifaddrs/freeifaddrs

leandro-lucarella-sociomantic committed:
linux/time: Add timer manipulation macros

joakim-brannstrom committed:
fix issue 16797 - Zero clock resolution lead to division by zero

jmdavis committed:
Merge pull request #1702 from joakim-brannstrom/fix16797

fix issue 16797 - Zero clock resolution lead to division by zero

somzzz committed:
Comment on try-catch issue

nemanja-boric-sociomantic committed:
Add TCP_* constants from Linux's netinet/tcp.h

nemanja-boric-sociomantic committed:
Add Linux-specific sched_setaffinity and sched_getaffinity

tramker committed:
Convert exception message to Windows console codepage

rainers committed:
Merge pull request #1687 from tramker/fixutf

Convert exception message to Windows console codepage

tramker committed:
Fixup PR #1687 - wrong buffer length used

rainers committed:
Merge pull request #1706 from tramker/pr1687fixup

Fixup PR #1687 - wrong buffer length used

WalterWaldron committed:
Revert object.hashOf changes from "use array interface to hashOf()"

fixes issues 16654 & 16764.

MartinNowak committed:
Merge pull request #1707 from WalterWaldron/fixHashOf

[REG 2.072] Revert object.hashOf changes from "use array interface to hashOf()"

andralex committed:
Merge pull request #1696 from somzzz/issue_5995

fix issue 5995 - string append negative integer causes segfault

schveiguy committed:
Merge pull request #1708 from dlang/stable

merge stable into master

kinke committed:
[MSVCRT] core.stdcpp.typeinfo: Don't implement exception ctors

Implementing them (wrongly btw, ignoring the msg) leads to an implicit call to the `super` ctor of base class `core.stdcpp.exception.std.exception`, which is only declared in D. Even if the ctor signature was correct (see below), it's mangled differently (see https://github.com/dlang/dmd/pull/5884), so there'd be an unresolved external anyway. So don't implement it, just declare it (also done for Glibc) to defer the linking error.

Quality-wise, the 2 `core.stdcpp` modules are at most at an alpha stage for MSVCRT. Other issues I've noted so far using **VS 2015**:
* The signature of the `core.stdcpp.exception.std.exception.this()` ctor is wrong. There's no default, i.e., parameter-less one, only `std::exception::exception(const char *_Message = "unknown", int x=1)`.
* `std::exception` only contains a pointer, no additional `bool` field.
* The signature of `core.stdcpp.typeinfo.std.type_info.name()` is wrong. It doesn't take any arguments.

The first 2 aren't crucial, as these C++ exceptions are most likely never constructed on the D side. But the last one will need to be fixed. @rainers: Which VS versions does DMD currently support?

Commit:
Merge pull request #1700 from nemanja-boric-sociomantic/ifaddrs

Add bindings for getifaddrs/freeifaddrs

Commit:
Merge pull request #1704 from nemanja-boric-sociomantic/tcp-cork

Add TCP_* constants from Linux's netinet/tcp.h

Commit:
Merge pull request #1697 from nemanja-boric-sociomantic/iconv

Add bindings for POSIX iconv library

Commit:
Merge pull request #1705 from nemanja-boric-sociomantic/addsched

Add Linux-specific sched_setaffinity and sched_getaffinity

yshui committed:
Fix missing bypassed chain in Error

UplinkCoder committed:
simplify _ArrayEq to make it work with the new ctfe engine

andralex committed:
Merge pull request #1712 from yshui/chain

Fix missing bypassed chain in Error

wilzbach committed:
Move changelog to a folder and use single file entries

dlang-bot committed:
Merge pull request #1716 from wilzbach/changelog-folder

Move changelog to a folder and use single file entries

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

wilzbach committed:
CircleCi: merge PRs with upstream branch

MartinNowak committed:
Merge pull request #1722 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable

andralex committed:
Merge pull request #1723 from wilzbach/use-merge-for-circleci

CircleCi: merge PRs with upstream branch

wilzbach committed:
CircleCi: use master as fallback for DMD

PetarKirov committed:
Add attributes core.thread part 1

This commit adds attributes to those functions
in `core.thread` that don't require changes
outside of the module.

Not much of this has profound effect to the users
but makes adding a templated `Thread`/`Fiber` classes
on the user's callback much easier later on,
which should help address the need for `@nogc` and
`nothrow` threading primitives.

PetarKirov committed:
Add attributes core.thread part 2

This commit fixes `Thread.~this()`, `thread_attachThis`
and partially `thread_entryPoint` and `thread_attachByAddrB`.

`thread_entryPoint` can't be made `@nogc` because it calls
`Thread.run`, `rt_moduleTlsCtor` and `rt_moduleTlsDtor`,
which can execute arbitrary user code.

`thread_attachByAddrB` can't be made `@nogc` because
it calls `thread_findByAddr`, which calls `Thread.opApply`,
which can't be `@nogc` because it calls a non-`@nogc`
delegate.

This change is a bit large because many call-sites needed
to be fixed in one go:

* On posix, `thread_entryPoint` calls `inheritLoadedLibraries`
and `cleanupLoadedLibraries`, and `Thread.start` calls
`rt.sections.pinLoadedLibraries` and `unpinLoadedLibraries`.

* `Thread.~this()` calls `rt.tlsgc.destroy`, which in turn
calls `rt.sections.finiTLSRanges`, which is implemented
in `rt.secions_{platform}`.

* Similarly, `thread_attachThis` calls `rt.tlsgc.init`, which in
turn calls `rt.sections.initTLSRanges`, which is implemented
in `rt.secions_{platform}`. On some platforms
`rt.sections.initTLSRanges` calls `rt.minfo.ModuleGroup.this`,
so I made it `nothrow` and `@nogc` too.

* Additionally, `rt.sections.initSections` and `finiSections` were
easy to fix, so I added appropriate attributes to them, even
though they are called only from `rt.dmain2.rt_init`.

Finally, some function from `rt.sections_elf_shared` was accessing
`ModuleInfo.name` which wasn't marked as `@nogc`, so I went
through `object.ModuleInfo` and made all of its members
`@nogc` (they were already `nothrow`).

PetarKirov committed:
Add attributes core.thread part 3

This commit adds attributes to Fiber.callImpl.

The call chain looks like this:
-> core.thread.Fiber.callImpl
  -> core.Fiber.switchIn
     -> core.Thread.pushContext/core.Thread.popContext
        -> core.thread.swapContext
           -> depending on the platform:
              -> rt.deh_win32/deh_win64_posix._d_eh_swapContext (On Windows and sometimes on Posix)
              -> rt.dwarfeh._d_eh_swapContextDwarf (on Posix)

PetarKirov committed:
Fix coding style in core.sync.mutex

PetarKirov committed:
Make core.sync.mutex.Mutex.this()/this(Object)/~this() nothrow and @nogc

PetarKirov committed:
Make `shared(core.sync.mutex.Mutex)` usable in `@safe` code

Additionally add `shared` overloads to:
+ core.sys.windows.winbase:
  * EnterCriticalSection
  * LeaveCriticalSection
  * TryEnterCriticalSection
+ core.sys.posix.pthread:
  * pthread_mutex_lock
  * pthread_mutex_unlock
  * pthread_mutex_trylock

Also add `tryLock_nothrow` version of `Mutex.tryLock`.

PetarKirov committed:
[WIP] Improve core.sync.mutex documentation

Commit:
Merge pull request #1699 from leandro-lucarella-sociomantic/libgen

Add missing POSIX <libgen.h> header

Commit:
Merge pull request #1701 from leandro-lucarella-sociomantic/timerfunc

posix/time: Add timer manipulation macros

MartinNowak committed:
add comment why this was written in such a way

dlang-bot committed:
Merge pull request #1713 from UplinkCoder/simplift_EqArray

simplify _ArrayEq to make it work with the new ctfe engine
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

carun committed:
linux/prctl: Added support for prctl.

carun committed:
linux/prctl: Updated makefiles.

carun committed:
linux/prctl: Updated review comments.

Commit:
Fix style in object.d file.

schveiguy committed:
Merge pull request #1729 from carun/master

linux/prctl: Added support for prctl.

jmdavis committed:
Merge pull request #1731 from sprinkle131313/fix-style-object

Fix style in object.d file.

MartinNowak committed:
Merge remote-tracking branch 'upstream/master' into stable

WalterBright committed:
compile with -dip1000

JackStouffer committed:
Revert "Qualify C-style memory allocations functions as pure"

This reverts commit 65d9a071b244611ca3973d79c8e64388af5c5439.

Commit:
Fix 17081 - Bodies in extern cpp functions in D files are not linked

PetarKirov committed:
[core.sync.mutex] Add a note to the docs about nothrow

PetarKirov committed:
[core.sync.mutex] Change assert(0, msg) to abort(msg); improve error messages

LemonBoy committed:
Always use same-type pointers when subtracting them

PetarKirov committed:
[core.sync.mutex] Cleanup existing unittest

PetarKirov committed:
[core.sync.mutex] Add a @safe ddoc-ed test and a @nogc nothrow one

PetarKirov committed:
[core.sync.mutex] Add a test for non-shared use of Mutex

andralex committed:
Merge pull request #1709 from kinke/patch-1

[MSVCRT] core.stdcpp.typeinfo: Don't implement exception ctors

andralex committed:
Merge pull request #1736 from Darredevil/issue-17081

Workaround Issue 17081 - Bodies in extern cpp functions in D files are not linked

JackStouffer committed:
Added pure back to alloca

andralex committed:
Merge pull request #1735 from JackStouffer/pure

Revert "Qualify C-style memory allocations functions as pure"

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

MartinNowak committed:
Merge pull request #1738 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable

andralex committed:
Merge pull request #1733 from WalterBright/dip1000

compile with -dip1000

DmitryOlshansky committed:
Merge pull request #1677 from Burgos/issue-11373

Add linux-specific `core.sys.linux.file` module

DmitryOlshansky committed:
Merge pull request #1737 from LemonBoy/correct-cast

Always use same-type pointers when subtracting them

WalterBright committed:
string.d: add return and scope

Commit:
Add missing const to iconv bindings

MartinNowak committed:
Merge pull request #1742 from Dicebot/inconv-const

Add missing const to iconv bindings

MartinNowak committed:
Merge pull request #1691 from rainers/nodefaultlib

Issue 12125: do not enforce specific MS C runtime

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

dlang-bot committed:
Merge pull request #1747 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

nordlow committed:
Add pureMalloc, pureCalloc and pureRealloc

somzzz committed:
array compare template function

WalterBright committed:
core.stdc.stdio: annotate with 'scope'

WalterBright committed:
core.stdc.stdlib: annotate with 'scope'

PetarKirov committed:
[core.sync.mutex] Abort if pthread_mutexattr_destroy fails

* Also make the error messages more descriptive

PetarKirov committed:
[core.sync.mutex] Address Andrei's comments

MartinNowak committed:
Merge pull request #1740 from WalterBright/string-scope

string.d: add return and scope

andralex committed:
Merge pull request #1726 from ZombineDev/add-attributes-to-core-thread

Add attributes to core.thread

andralex committed:
Merge pull request #1728 from ZombineDev/add-attributes-to-core-sync-mutex

Makes core.sync.mutex.Mutex usable in modern code

nemanja-boric-sociomantic committed:
Add bindings for POSIX getdelim and getline

WalterBright committed:
[scope] update core.demangle for 'scope' mangling

WalterBright committed:
Merge pull request #1748 from WalterBright/scope-stdio

core.stdc.stdio: annotate with 'scope'

WalterBright committed:
[scope] add 'scope' to delegates

WalterBright committed:
[scope] add 'scope' to delegates

andralex committed:
Merge pull request #1746 from nordlow/pure-c-allocs

Add pure wrappers for C style allocation

rainers committed:
fix pureMalloc test for Win32

CyberShadow committed:
Merge pull request #1754 from rainers/fixMaster

fix pureMalloc test for Win32

kinke commented:
Thanks for that, but upstreaming all those NetBSD fixes to official dlang/druntime makes much more sense, so that all D compilers benefit.

WalterBright committed:
Merge pull request #1752 from WalterBright/scope-dmain2

[scope] add 'scope' to delegates

WalterBright committed:
Merge pull request #1751 from WalterBright/scope-demangle

[scope] update core.demangle for 'scope' mangling

WalterBright committed:
make toString() @safe in core.sys.windows.stacktrace

MartinNowak committed:
Merge pull request #1753 from WalterBright/scope-gc

[scope] add 'scope' to delegates

WalterBright committed:
[scope] add annotations to core.internal.string

WalterBright committed:
add _d_assertp, _d_unittestp, _d_arrayboundsp

WalterBright committed:
Merge pull request #1761 from WalterBright/assertp

add _d_assertp, _d_unittestp, _d_arrayboundsp

andralex committed:
Merge pull request #1757 from WalterBright/safe-stacktrace

make toString() @safe in core.sys.windows.stacktrace

andralex committed:
Merge pull request #1759 from WalterBright/string-return

[scope] add annotations to core.internal.string

WalterBright committed:
line_trace.exp: make more resistant to 'scope' changes

rainers committed:
scan DATA and TLS according to compiler generated pointer infos

andralex committed:
Merge pull request #1762 from WalterBright/line_exp

line_trace.exp: make more resistant to 'scope' changes

andralex committed:
Merge pull request #1749 from WalterBright/scope-stdlib

core.stdc.stdlib: annotate with 'scope'

rainers committed:
implement precise scan DATA and TLS for MS-COFF builds

schveiguy committed:
Push back deprecation of cycle detection one more major release

wilzbach committed:
Set bootstrap DMD version to 2.072.2

wilzbach committed:
Add fallback for CodeCov

MartinNowak committed:
Merge pull request #1764 from dlang/schveiguy-patch-1

Push back deprecation of bad cycle detection one more major release

schveiguy committed:
Update handling of previously undetected cycles to abort execution.

wilzbach committed:
Update changelog README

WalterBright committed:
Merge pull request #1675 from Burgos/fix-overflow

Add some more overflow checks inside array allocation

WalterBright committed:
Merge pull request #1767 from wilzbach/update-changelog-readme

Update changelog README

WalterBright committed:
Merge pull request #1765 from schveiguy/updatecyclehandling

Update handling of previously undetected cycles to abort execution.

rainers committed:
OMF: move pointer reference data to CODE segment

rainers committed:
windows: disable precise scanning of DATA/TLS

MartinNowak committed:
Merge pull request #1763 from rainers/data_ptrref

scan DATA and TLS according to compiler generated pointer infos

WalterBright committed:
workaround sscanf bug in old snn.lib

MartinNowak committed:
use correct version CRuntime_DigitalMars

MartinNowak committed:
Merge pull request #1768 from WalterBright/sscanf-workaround

workaround sscanf bug in old snn.lib

tomer-weka committed:
Fix wrong version name in core.sys.linux.ifaddrs

dlang-bot committed:
Merge pull request #1769 from tomerfiliba/ifaddrs-fix

Fix wrong version name in core.sys.linux.ifaddrs
merged-on-behalf-of: David Nadlinger <code@klickverbot.at>

nemanja-boric-sociomantic committed:
Change version(Linux) to version(linux)

WalterBright committed:
Merge pull request #1771 from nemanja-boric-sociomantic/version-linux

Fix remaining `version(Linux)` to `version(linux)`

CyberShadow committed:
core.sys.linux.sched: Fix DDoc syntax

WalterBright committed:
Merge pull request #1775 from CyberShadow/pull-20170226-043259

core.sys.linux.sched: Fix DDoc syntax

MartinNowak committed:
adding Jenkinsfile to hook with new ci2.dawg.eu instance

- should transfer the ownership of the ci repo to dlang/ci

Commit:
Update windows console text attributes

RazvanN7 committed:
Updated makefile to use generated/os/release/model/dmd

dlang-bot committed:
Merge pull request #1778 from mogud/patch-1

Add missing Windows CHAR_INFO attributes
merged-on-behalf-of: David Nadlinger <code@klickverbot.at>

andralex committed:
Merge pull request #1779 from RazvanN7/Makefile_change

Updated makefile to use generated/os/release/model/dmd

schveiguy committed:
Remove unused rt.typeinfo.ti_AC

dlang-bot committed:
Merge pull request #1782 from schveiguy/removeti_ac

Remove unused rt.typeinfo.ti_AC
merged-on-behalf-of: Steven Schveighoffer <schveiguy@users.noreply.github.com>

somzzz committed:
fixed nits & added tests & dstrcmp

CyberShadow committed:
.gitignore: Fix Makefile rule

The introduction of the .gitignore rule lacked any description or
rationale (PR #682), so I'm guessing this is to allow people to create
a local Makefile so that `make` works without needing to specify `-f
posix.mak`.

Fix this rule so that it doesn't apply to the actually versioned
Makefile files under the test directory.

CyberShadow committed:
Add tests for class/struct/array comparisons

PR #1781 is moving array comparisons to a new template; the initial
implementation included changes which may have introduced performance
and behavior regressions.

Add explicit tests for these cases to detect such regressions.

MartinNowak committed:
Merge pull request #1784 from CyberShadow/pull-20170304-141153

Add tests for class/struct/array comparisons

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

# Conflicts:
#	src/rt/minfo.d

wilzbach committed:
Remove dmd.conf target from CircleCi

somzzz committed:
fixed breaking change & code adjustments

dlang-bot committed:
Merge pull request #1785 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

somzzz committed:
Comments and nits

MartinNowak committed:
Merge pull request #1725 from wilzbach/circleci-master-fallback

CircleCi: use master as fallback for DMD

MartinNowak committed:
Merge pull request #1777 from dlang/Jenkinsfile

adding Jenkinsfile to hook with new ci2.dawg.eu instance

andralex committed:
Merge pull request #1781 from somzzz/adCmpT

Array Compare Template Function

andralex committed:
One more pass through __cmp

andralex committed:
Factor in implementation for class and interface types

andralex committed:
Fix __cmp for object types, was not even called

andralex committed:
Fix copypasta mistake

Temtaime committed:
Add missing function to iconv.

dlang-bot committed:
Merge pull request #1750 from nemanja-boric-sociomantic/getline

Add bindings for POSIX getdelim and getline
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

Temtaime committed:
Update iconv.d

andralex committed:
Preserve legacy weird behavior by which structs and unions in arrays were compared with memcmp

andralex committed:
Further reduction of code

andralex committed:
Aw what the heck implemented reuse of implementation for all builtin types

andralex committed:
Oooops...

andralex committed:
Merge pull request #1787 from andralex/__cmp

One more pass through __cmp

jmdavis committed:
Small code simplification in core.time.

We already have the type with Args[i], so there's no need to ask args[i] for
its type.

MartinNowak committed:
run pipeline outside of node block

- to avoid executor startvation, see Dicebot/dlangci#30

MartinNowak committed:
Merge pull request #1790 from MartinNowak/jenkins

run pipeline outside of node block

MartinNowak committed:
Merge pull request #1789 from jmdavis/time

[trivial] Small code simplification in core.time.

MartinNowak committed:
Merge pull request #1786 from wilzbach/remove-dmd-conf

Remove dmd.conf target from CircleCi

MartinNowak committed:
fix Jenkinsfile

MartinNowak committed:
Merge pull request #1791 from MartinNowak/jenkins

fix Jenkinsfile

MartinNowak committed:
trailing ws

dlang-bot committed:
Merge pull request #1492 from nrTQgc/netbsd_patch

NetBSD path
merged-on-behalf-of: David Nadlinger <code@klickverbot.at>

dlang-bot committed:
Merge pull request #1788 from Temtaime/patch-1

Add missing function to iconv.
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

joakim-noah commented:
Is this all in?  Please close PRs that have been merged upstream already.
